numba/tests/test_parfors.py

```diff
@@ -1596,9 +1596,10 @@ def test_vectorizer_fastmath_asm(self):
         # packed dtype=double add and sqrt instructions.
         def will_vectorize(A):
             n = len(A)
+            acc = 0
             for i in range(n):
-                A[i] += np.sqrt(i)
-            return A
+                acc += np.sqrt(i)
+            return acc
```

Comment:
Is this change needed because the new LLVM6 will always vectorize the code, and, reduction-loop will only be vectorized if fastmath is ON?

Reply:
I think that is the case. The LLVM 6 vectorizer seems to be able to vertically vectorize more loops. This change forces the need for associativity to be broken to vectorize, so should only work under `fastmath`.